scripts/zkai_attention_derived_d128_native_mlp_proof_route_gate.py

```
@@ -372,7 +382,23 @@ def build_context() -> dict[str, Any]:
         raise NativeMlpProofRouteError("derived native activation proof backend version drift")
     if native_activation_envelope.get("decision") != "GO_D128_ACTIVATION_SWIGLU_AIR_PROOF":
         raise NativeMlpProofRouteError("derived native activation proof decision drift")
+    if _dict(native_down_envelope.get("input"), "derived native down envelope input") != native_down:
+        raise NativeMlpProofRouteError("derived native down envelope/input mismatch")
+    if native_down_envelope.get("proof_backend_version") != "stwo-d128-down-projection-air-proof-v1":
+        raise NativeMlpProofRouteError("derived native down proof backend version drift")
+    if native_down_envelope.get("decision") != "GO_D128_DOWN_PROJECTION_AIR_PROOF":
+        raise NativeMlpProofRouteError("derived native down proof decision drift")
+    if native_down.get("source_hidden_activation_commitment") != native_activation.get("hidden_activation_commitment"):
+        raise NativeMlpProofRouteError("derived native activation/down hidden commitment mismatch")
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)